src/Analyzers/Core/Analyzers/RemoveUnnecessarySuppressions/SuppressMessageAttributeState.cs

```diff
@@ -53,22 +54,18 @@ public SuppressMessageAttributeState(Compilation compilation, INamedTypeSymbol s
             CancellationToken cancellationToken,
             out ImmutableArray<(string name, IOperation value)> namedAttributeArguments)
         {
-            var attribute = model.GetOperation(attributeSyntax, cancellationToken);
-            if (attribute == null)
+            var operation = (model.GetOperation(attributeSyntax, cancellationToken) as IAttributeOperation)?.Operation;
+            if (operation is not IObjectCreationOperation { Initializer: { } initializerOperation })
             {
                 namedAttributeArguments = ImmutableArray<(string name, IOperation value)>.Empty;
                 return false;
             }
 
-            // Workaround for https://github.com/dotnet/roslyn/issues/18198
-            // Use 'IOperation.Children' to get named attribute arguments.
-            // Each named attribute argument is represented as an 'ISimpleAssignmentOperation'
-            // with a constant value assignment to an 'IPropertyReferenceOperation' in the operation tree.
             using var _ = ArrayBuilder<(string name, IOperation value)>.GetInstance(out var builder);
-            foreach (var childOperation in attribute.ChildOperations)
+            foreach (var initializer in initializerOperation.Initializers)
             {
-                if (childOperation is ISimpleAssignmentOperation simpleAssignment &&
-                    simpleAssignment.Target is IPropertyReferenceOperation propertyReference &&
+                var simpleAssignment = (ISimpleAssignmentOperation)initializer;
```

Comment:
@333fred I couldn't find a case where this can fail. I can switch to `is` check but want to make sure we have a good coverage if there is a possibility that `initializer` is not `ISimpleAssignmentOperation`.